src/spec-node/singleContainer.ts

```diff
@@ -126,27 +126,53 @@ async function buildAndExtendImage(buildParams: DockerResolverParameters, config
 	const { config } = configWithRaw;
 	const dockerfileUri = getDockerfilePath(cliHost, config);
 	const dockerfilePath = await uriToWSLFsPath(dockerfileUri, cliHost);
+	const buildContextPath = await uriToWSLFsPath(getDockerContextPath(cliHost, config), cliHost);
 	if (!cliHost.isFile(dockerfilePath)) {
 		throw new ContainerError({ description: `Dockerfile (${dockerfilePath}) not found.` });
 	}
 
 	let dockerfile = (await cliHost.readFile(dockerfilePath)).toString();
 	const originalDockerfile = dockerfile;
 	let baseName = 'dev_container_auto_added_stage_label';
-	if (config.build?.target) {
-		// Explictly set build target for the dev container build features on that
-		baseName = config.build.target;
+	let imageBuildInfo: ImageBuildInfo;
+
+	if (config.prebuildDockerfile) {
+		const intermediateBuildAargs = ['build', '-f', dockerfilePath, '-t'];
+		let intermediateImageName = 'vsc_tmp_' + cliHost.path.basename(buildContextPath);
+		intermediateBuildAargs.push(intermediateImageName);
+		intermediateBuildAargs.push(buildContextPath);
```

Comment:
`--target` from the devcontainer.json needs to apply here like below.

Should we add `--platform`, `--push`, `--output`, `--cache-from`, `--no-cache`, `--pull`, `--build-arg` and `BUILDKIT_INLINE_CACHE=1` depending on parameters like below?

Makes we wonder if we should instead of "prebuild the user's Dockerfile" turn the change around and "add features later".

Reply:
Noted on the build args

> Makes we wonder if we should instead of "prebuild the user's Dockerfile" turn the change around and "add features later".

I don't quite get what you mean here, do you mind elaborating how the features would be added later?